packages/core/src/app-middlewares/before/fetch-stashed-bundle.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+const fetch = require('../../tools/fetch');
+const _ = require('lodash');
+const { StashedBundleError } = require('../../errors');
+const crypto = require('crypto');
+// Use the fernet npm package for proper Fernet token handling
+const fernet = require('fernet');
+
+const withRetry = async (fn, retries = 3, delay = 100, attempt = 0) => {
+  try {
+    return await fn();
+  } catch (error) {
+    if (attempt >= retries) {
+      throw error;
+    }
+
+    await new Promise((resolve) => setTimeout(resolve, delay));
+    return withRetry(fn, retries, delay, attempt + 1);
+  }
+};
+
+/**
+ * Decrypt a bundle using secret key
+ *
+ * This matches the backend implementation:
+ * 1. Hash the secret with SHA256 to get 32 bytes
+ * 2. Base64url encode those bytes to make Fernet-compatible key
+ * 3. Use Fernet library to decrypt (handles all token parsing internally)
+ *
+ * @param {string} fernetToken - The Fernet token to decrypt
+ * @param {string} secret - The secret key for decryption
+ * @returns {Object} The decrypted bundle object
+ */
+const decryptBundleWithSecret = (fernetToken, secret) => {
```

Comment:
Cool! Wonder if this is worth moving to a util function? But nbd, since we don't have any other encrypt/decrypt